### PR TITLE
Automate Protected Devices to run `particle serial` commands

### PR DESCRIPTION
**Pull request by keeramis**
## Description

Support Protected Devices while running `particle serial xxx` commands

## How to Test

Run `particle serial xxx` commands on your Protected Devices as well as Open Devices

## Related Issues / Discussions

https://app.shortcut.com/particle/story/129645/automate-protected-devices-to-run-particle-serial-commands
https://app.shortcut.com/particle/story/129646/automate-protected-devices-to-run-particle-serial-commands-e2e-tests

## Completeness

- [x] User is totes amazing for contributing!
- [ ] Contributor has signed [CLA](https://docs.google.com/a/particle.io/forms/d/1_2P-vRKGUFg5bmpcKLHO_qNZWGi5HKYnfrrkd-sbZoA/viewform)
- [ ] Problem and solution clearly stated
- [ ] Tests have been provided
- [ ] Docs have been updated
- [ ] CI is passing

### CODE REVIEW

**Comment by monkbroc** [comment]
The serial monitor command hangs after I press reset on an Argon.
```
 particle.js serial monitor
Opening serial monitor for com port: " /dev/ttyACM0 "
Serial monitor opened successfully:

Serial connection closed.
(hangs here)
```

**Comment by monkbroc** [comment]
With `particle serial monitor --follow`, it doesn't put the device in service mode after every reconnect so it's possible to get lose the logs after a bunch of resets.

```
particle serial monitor --follow
Polling for available serial device...Opening serial monitor for com port: " /dev/ttyACM0 "
Serial monitor opened successfully:

Serial connection closed.  Attempting to reconnect...
Serial monitor opened successfully:
0000001674 [ncp.esp32.at] TRACE: > AT
0000001676 [ncp.esp32.at] TRACE: < OK
0000002676 [ncp.esp32.client] TRACE: NCP ready to accept AT commands
0000002677 [ncp.esp32.at] TRACE: > AT+MVER
0000002678 [ncp.esp32.at] TRACE: < 5
0000002678 [ncp.esp32.at] TRACE: < OK
0000002679 [ncp.esp32.at] TRACE: > AT+GETMAC=0
0000002681 [ncp.esp32.at] TRACE: < +GETMAC: "30:ae:a4:d2:b2:64"
0000002681 [ncp.esp32.at] TRACE: < OK
0000002683 [ncp.esp32.at] TRACE: > AT+CMUX=0
0000002690 [ncp.esp32.at] TRACE: < OK
0000002690 [ncp.esp32.mux] INFO: Starting GSM07.10 muxer
0000002691 [ncp.esp32.mux] INFO: Opening mux channel 0
0000002692 [ncp.esp32.mux] INFO: GSM07.10 muxer thread started
0000002695 [ncp.esp32.mux] INFO: Opening mux channel 1
0000002699 [ncp.esp32.at] TRACE: > AT
0000002702 [ncp.esp32.at] TRACE: < OK
0000002702 [ncp.esp32.at] TRACE: > AT+CWDHCP=0,3
0000002707 [ncp.esp32.at] TRACE: < OK
0000002707 [ncp.esp32.client] TRACE: NCP state changed: 1
0000002707 [net.esp32ncp] TRACE: NCP event 1
0000002708 [net.esp32ncp] TRACE: NCP event 3
0000002708 [net.esp32ncp] TRACE: NCP power state changed: IF_POWER_STATE_UP
0000002709 [system.nm] TRACE: Interface 4 power state changed: UP
0000002715 [ncp.esp32.at] TRACE: > AT+CWLAP

Serial connection closed.  Attempting to reconnect...
Serial monitor opened successfully:

```

**Comment by monkbroc** [comment]
Serial monitor without follow now properly exits. Good fix :+1: 

I'm getting a crash when pressing reset on an Argon during serial monitor follow.
```
$ particle.js serial monitor --follow
Polling for available serial device...Opening serial monitor for com port: " /dev/ttyACM0 "
Serial monitor opened successfully:

Serial connection closed.  Attempting to reconnect...
TypeError: Cannot read properties of undefined (reading 'getDevice')
```

**Review by monkbroc — APPROVED** [review]
After the fix for serial monitor --follow, this is good to merge :+1:

File: src/cmd/serial.js [hunk 1]
```diff
@@ -211,7 +225,31 @@ module.exports = class SerialCommand extends CLICommandBase {
 		}
 
 		function reconnect(){
```
Comment (keeramis) [inline src/cmd/serial.js:227]: @monkbroc Could you kindly take a look at this refactor and let me know if you see this is not ok for merge?

Once the `serial monitor --follow` command is reconnecting to the device, the usb handle is unable to be obtained via `usbUtils.getOneUsbDevice` quickly enough so I added an arbitrary delay. Tested on Argon and it takes a few milliseconds (have not tested yet on P2 coz unfortunately my P2 is at work location and I will get it later today). Picked 10 ms without solid reasoning.

**Review by hugomontero — APPROVED** [review]
I was able to test it with a P2 with `6.1.1` device os version
After several resets (button reset, command reset) the monitor still showing log info

image

**Review by monkbroc — APPROVED** [review]

**Pull request merged**